test/settings/settings_test.cpp

```diff
@@ -41,11 +41,45 @@ class SettingsTests : public TerrierTest {
   void TearDown() override { delete db_main_; }
 
   static void EmptySetterCallback(const std::shared_ptr<common::ActionContext> &action_context UNUSED_ATTRIBUTE) {}
+
+  /**
+   * This is a hacky placeholder variable that we can use in lambda callbacks
+   * If you use this in the tests below, then you have to make sure that reset it before
+   * you use it.
+   */
+  inline static bool invoked_;  // NOLINT
```

Comment:
This is only used in SetterCallbackTest for now, so you can probably just move it to inside that test case and pass it by reference to the lambda.

Reply:
I couldn't figure out how to get this work through a lambda because it has to match the `settings::setter_callback_fn` signature.

Reply:
I see, that's fine then.